### ISSUE TITLE
[Feature Request] Download to network share

### ISSUE THREAD

**Issue by 5p0ng3b0b** [recloudstream/cloudstream#474]
### Describe your suggested feature

It is not really practical to download on devices like firesticks due to limited internal storage and no provision for external storage. Can the app be adapted to provide an option to set the download path to an NFS or SMB share?

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by boognish-rising** [comment]
Just FYI, I'm reasonable certain that all Fire devices (Firestick/cube/TV/OS/4K/Tablets) have OTG functionality. While I can't personally vouch for every kind of Fire device ever released, I can vouch/confirm w 100% certainty that it _is_ true of Firesticks, FireTV/OS (Amazon branded FireTV as well as Hisense Hisense Smart TV (FireTV at the helm); but all Fire devices are essentially the same, i.e., they're all running FireOS. )and Fire tablets. Specific to your situation with a Fire stick, I've never encountered an OTG cable/hub, etc that _didn't_ work and allow for expanded storage but I've seen many that seem to be specifically made for the purpose. If you have one on hand, give it a shot and if it's a no go, maybe try to err on the side of caution and just get one that expressly mentions Firestick compatibility. FYI-not sure if you have any USB type C hubs available but most have a PD input, and while not always specifically advertised as such, 9.9 times out of 10 these will work perfectly. I have one connected to a Chromecast 4k with GoogleTV and a FireOS Hisense smart tv (w/ Type-C→A adapter) but prior to getting the Hisense FireTV, I had one connected to a 4K Firestick.

OTG will allow for storage expansion, not 90 FS formatting and I seem to remember something about it only taking SD cards but I'm being any type of external storage medium will work as long as it's formatted correctly.

That said, I'm 100% on board with the suggested enhancement; it would take an extra step out of the process for me (either physically changing location of the storage device or using a 3rd party file manager with network storage functionality (X-plore/Ghost Commander, etc) but only after first DLing to SAF (the former option actually involves 2 extra steps due to what I believe is a bug with SAF access when selecting custom DL locations(I can see my connected USB storage device but can't select it. A connected mouse (sort of) allows me to select the location however A) the cursor isn't visible so to say this is an infuriating task is putting it mildly and B) after to select the storage device, I can't for the life of me actually click select. Not with the remote, not with a mouse (cursor still invisible but it's in the bottom right corner so theoretically, it should be a walk in the park compared to selecting a storage device but I made A LOT of attempts and ultimately had to give up for fear that I would destroy the mouse or TV or both in a furious rage

**Comment by boognish-rising** [comment]
[CIFS Documents Provider](https://github.com/wa2c/cifs-documents-provider)

If "Network Share" = SMB (or CiFS? Which I've never heard of apart from this app), this app is awesome. 

**[Features]**

* Access to Windows shared folders via Storage Access Framework (SAF).
* Read, write, and directory access in shared folders from external applications by using SAF.
* File sharing from shared folder.
* File transfer any files.
* Multiple connection settings can be saved.

It's not quite what you're requesting but it's pretty close and either way, the best I could find. If you have a solid internet connection, the performance should probably be identical

**Comment by 5p0ng3b0b** [comment]
cheers that'll do 👍
